### ISSUE TITLE
Handle unsupported content format on notification correctly

### ISSUE THREAD

**Issue by joakimeriksson** [eclipse/leshan#244]
I am running a leshan server and a Contiki based LWM2M client and when I do an observation on for example sensor values I get exceptions as soon as the notification comes in to leshan. It is related to a missing null pointer check at line 245 org.eclipse.leshan.server.californium.impl.ObservationServiceImpl.onNotification

Where there is a piece of code like this:
```
 // decode response
                List<TimestampedLwM2mNode> timestampedNodes = decoder.decodeTimestampedData(coapResponse.getPayload(),
                        contentFormat, observation.getPath(), model);

                // create lwm2m response
                ObserveResponse response;
                if (timestampedNodes.size() == 1 && !timestampedNodes.get(0).isTimestamped()) {
                    response = new ObserveResponse(ResponseCode.CONTENT, timestampedNodes.get(0).getNode(), null,
                            observation, null, coapResponse);
                } else {
                    response = new ObserveResponse(ResponseCode.CONTENT, null, timestampedNodes, observation, null,
                            coapResponse);
                }

```
This code assumes that decoder will always return an object with timestamped nodes but I guess it is not doing that in some cases. (line 245 is the if (timestamped...).

The exception I get is the following:
```

Jan 13, 2017 8:43:57 AM org.eclipse.californium.core.network.stack.BlockwiseLayer$BlockCleanupTask run
INFO: Block2 transfer timed out: {1}
Jan 13, 2017 8:43:57 AM org.eclipse.californium.core.network.CoapEndpoint$5 run
SEVERE: Exception in protocol stage thread: null
java.lang.NullPointerException
	at org.eclipse.leshan.server.californium.impl.ObservationServiceImpl.onNotification(ObservationServiceImpl.java:245)
	at org.eclipse.californium.core.network.CoapEndpoint$NotificationDispatcher.onNotification(CoapEndpoint.java:568)

```

**Comment by joakimeriksson** [comment]
Found out the reason - I was still sending old plain-text format (1541). Maybe there could be an improved exception for this!?

**Comment by sbernard31** [comment]
Yes you're right we need to provide more feedback when an error occur on notification, there is an issue about that https://github.com/eclipse/leshan/issues/206

We should handle unsupported format in better way than just returning null, maybe by rising an exception ? WDYT ?

**Comment by joakimeriksson** [comment]
Good improvement - this is how all things should be handled to provide useful feedback to anyone using  Leshan! Many thanks!

**Comment by sbernard31** [comment]
Fixed since #245 was integrated in master
